.github/workflows/main.yml

```diff
@@ -24,7 +24,7 @@ jobs:
         name: cargo test
         with:
           command: test
-          args: --all-targets --benches
+          args: --all-targets --benches -- --test-threads=1
```

Comment:
Hmm, don't need to do it here but I think we should move all test related things to a bash script so that users can easily run what gets run on CI. I know that became quite a struggle for rust-lightning, but it's quite nice to have everything in a single `.sh` that the CI also ends up calling.